From 6b8ae289faefc9f5541aa36b2f5ae876c96f1139 Mon Sep 17 00:00:00 2001
From: Ding Wei <leo.ding@rock-chips.com>
Date: Fri, 22 May 2020 09:49:47 +0800
Subject: [PATCH] video: rockchip: mpp: fix error for pop twice

when wait time out twice, it may wait recursion, and then
trigger pop twice.

Change-Id: I6c689f4267e0c03e8cf1e41d68796a6d042afce7
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 drivers/video/rockchip/mpp/mpp_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/mpp/mpp_common.c b/drivers/video/rockchip/mpp/mpp_common.c
index 91ca4d194d29..b3df82f293a9 100644
--- a/drivers/video/rockchip/mpp/mpp_common.c
+++ b/drivers/video/rockchip/mpp/mpp_common.c
@@ -699,7 +699,7 @@ static int mpp_wait_result(struct mpp_session *session,
 				session, task);
 			ret = -ETIMEDOUT;
 		} else {
-			ret = mpp_wait_result(session, msgs);
+			return mpp_wait_result(session, msgs);
 		}
 	}
 
-- 
2.35.3

